server/models/estimationFeatureModel.js

```diff
@@ -126,5 +126,54 @@ estimationFeatureSchema.statics.addFeatureByEstimator = async (featureInput, est
 }
 
 
+estimationFeatureSchema.statics.updateFeatureByEstimator = async (featureInput, estimator) => {
+    validate(featureInput, estimationEstimatorUpdateFeatureStruct)
+    if (!estimator || !userHasRole(estimator, SC.ROLE_ESTIMATOR))
+        throw new AppError('Not an estimator', EC.INVALID_USER, EC.HTTP_BAD_REQUEST)
+    let estimation = await EstimationModel.findById(featureInput.estimation._id)
+    if (!estimation)
+        throw new AppError('Estimation not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    if (!_.includes([SC.STATUS_ESTIMATION_REQUESTED, SC.STATUS_CHANGE_REQUESTED], estimation.status))
+        throw new AppError("Estimation has status as ["+estimation.status+"]. Estimator can only update feature into those estimations where status is in [" + SC.STATUS_ESTIMATION_REQUESTED + ", " + SC.STATUS_CHANGE_REQUESTED + "]", EC.INVALID_OPERATION, EC.HTTP_BAD_REQUEST)
+
+    let estimationFeature = await EstimationFeatureModel.findById(featureInput._id)
+    if(!estimationFeature)
+        throw new AppError('Estimation feature not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    if (estimationFeature.repo && estimationFeature.repo._id) {
+        // Feature is added from repository
+        // find if Feature actually exists there
+        let repositoryFeature = await RepositoryModel.findById(estimationFeature.repo._id)
+        if (!repositoryFeature)
+            throw new AppError('Feature not part of repository', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+        if (_.includes([SC.STATUS_APPROVED], repositoryFeature.status))
```

Comment:
Feature into repository can only be updated if that feature was added from this estimation only. For this estimation._id of RepositoryFeature can be compared with estimation id

Reply:
Also if you have added updateFeature method in RepositoryModel than you should be using that method to update feature rather than directly updating from here